### PR TITLE
Refactor of cosmos modules

### PR DESCRIPTION
**Pull request by krhubert**
related to https://github.com/mesg-foundation/engine/issues/1639

TODO: 
- use one function in reset/cli querier
- Add cosmos wrap error to all module errors

### CODE REVIEW

File: cosmos/module_client.go [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+	var out *instancepb.Instance
+	route := sroutef("%s/%s/%s", instance.QuerierRoute, instance.QueryGetInstance, hash)
+	return out, mc.QueryJSON(route, nil, &out)
+}
+
+// ListInstance returns all instances.
+func (mc *ModuleClient) ListInstance(req *api.ListInstanceRequest) ([]*instancepb.Instance, error) {
+	var out []*instancepb.Instance
+	route := sroutef("%s/%s", instance.QuerierRoute, instance.QueryListInstances)
+	return out, mc.QueryJSON(route, req, &out)
```
Comment (NicolasMahe) [inline cosmos/module_client.go]: Only the filter should be passed to the query

original: https://github.com/mesg-foundation/engine/pull/1650/files#diff-cfb2b44e2ea93e1c0e4199d390e79f83L31
```diff
-	if err := s.client.QueryJSON(fmt.Sprintf("custom/%s/%s", instance.QuerierRoute, instance.QueryListInstances), request.Filter, &instances); err != nil {	
```

Comment (krhubert) [reply cosmos/module_client.go]: Strange it should panic, but I've fixed it.

File: sdk/execution/sdk.go [hunk 2]
```diff
@@ -1,118 +0,0 @@
...
-		client:     client,
-		serviceSDK: serviceSDK,
-		runnerSDK:  runnerSDK,
-	}
-	return sdk
-}
-
-// Create creates a new execution.
-func (s *SDK) Create(req *api.CreateExecutionRequest) (*executionpb.Execution, error) {
-	execution.M.Created.Add(1)
```
[the remark this answers is not in the document]
Comment (NicolasMahe) [reply sdk/execution/sdk.go]: could be nice to have a define a shared metric object and use it across the engine
but then it's against the client / module separation

[the remark this answers is not in the document]
Comment (NicolasMahe) [reply sdk/execution/sdk.go]: i'm ok to merge without this 👍

[the remark this answers is not in the document]
Comment (krhubert) [reply sdk/execution/sdk.go]: They should be right im the execution. Chęci keeper się. Unfournetly i coundnt find a good place for them. We can add that to refactor list to keep track of that

**Pull request merged**